### ISSUE TITLE
thread 'main' panicked at 'assertion failed: pts.len() >= 2', geom/src/polyline.rs:27:9

### ISSUE THREAD

**Issue by jraymakers** [dabreegster/abstreet#72]
I was working on the VIP challenge when the game suddenly disappeared.  I think I was jumping to a time and was either just about to or had just clicked the "stop" button.

The last few lines the console log were:

```
1 warnings:
Points changed angles from Angle(45.63918808384173 degrees) to Angle(41.041700251198634 degrees) during polyline shifting
1 warnings:
Points changed angles from Angle(147.62541778612044 degrees) to Angle(102.67117209846786 degrees) during polyline shifting
1 warnings:
Points changed angles from Angle(45.63914797679607 degrees) to Angle(41.82078534270572 degrees) during polyline shifting
1 warnings:
Points changed angles from Angle(147.62110914350106 degrees) to Angle(102.66557430080944 degrees) during polyline shifting
1 warnings:
Points changed angles from Angle(45.639554776830614 degrees) to Angle(42.33145689592787 degrees) during polyline shifting
thread 'main' panicked at 'assertion failed: pts.len() >= 2', geom/src/polyline.rs:27:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
********************************************************************************
UI broke! Primary sim:
********************************************************************************
At 09:35:45.5
Wrote ../data/player/camera_state/montlake.json
```

I was running a built-from-source version using "cargo run --release" on Mac.

Attached is the full console log and the camera state JSON file (which last line of the console log referenced).

[abstreet_console_output.txt](https://github.com/dabreegster/abstreet/files/4817122/abstreet_console_output.txt)

[montlake.json.txt](https://github.com/dabreegster/abstreet/files/4817125/montlake.json.txt)

**Comment by dabreegster** [comment]
`RUST_BACKTRACE=1 cargo run --release` would have been helped; I updated the docs. I tried reproing by focusing the camera in the same place, but no luck so far.

There are lots of places that could potentially trip this assertion; related to #71, I'm going to do a full sweep of the code and be much more careful about panics. This was probably caused by trying to draw a funny shaped arrow or something; gracefully failing by not drawing the arrow would be much better than crashing.
